src/ILCompiler.RyuJit/src/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -484,5 +485,109 @@ private InfoAccessType constructStringLiteral(CORINFO_MODULE_STRUCT_* module, md
             ppValue = (void*)ObjectToHandle(stringObject);
             return stringObject.RepresentsIndirectionCell ? InfoAccessType.IAT_PVALUE : InfoAccessType.IAT_VALUE;
         }
+
+        private void setVars(CORINFO_METHOD_STRUCT_* ftn, uint cVars, NativeVarInfo* vars)
+        {
+            if (_localSlotToInfoMap == null && _parameterIndexToNameMap == null)
+            {
+                return;
+            }
+
+            uint paramCount = (_parameterIndexToNameMap == null) ? 0 : (uint)_parameterIndexToNameMap.Count;
+            Dictionary<uint, DebugVarInfo> debugVars = new Dictionary<uint, DebugVarInfo>();
+            int i;
+
+            for (i = 0; i < cVars; i++)
+            {
+                NativeVarInfo nativeVarInfo = vars[i];
+
+                if (nativeVarInfo.varNumber < paramCount)
+                {
+                    string name = _parameterIndexToNameMap[nativeVarInfo.varNumber];
+                    updateDebugVarInfo(debugVars, name, true, nativeVarInfo);
+                }
+                else if (_localSlotToInfoMap != null)
+                {
+                    ILLocalVariable ilVar;
+                    uint slotNumber = nativeVarInfo.varNumber - paramCount;
+                    if (_localSlotToInfoMap.TryGetValue(slotNumber, out ilVar))
+                    {
+                        updateDebugVarInfo(debugVars, ilVar.Name, false, nativeVarInfo);
+                    }
+                }
+            }
+
+            i = 0;
+            _debugVarInfos = new DebugVarInfo[debugVars.Count];
+            foreach (var debugVar in debugVars)
+            {
+                _debugVarInfos[i] = debugVar.Value;
+                i++;
+            }
+        }
+
+        // Create a DebugLocInfo which is a table from native offset to source line.
```

Comment:
Nit: consider upgrading this to an XML method comment.

Reply:
Done